### ISSUE TITLE
Servername in header

### ISSUE THREAD

**Issue by davidak** [mholt/caddy#738]
I want to do something like

```
header / X-Backend-Server {hostname}
```

where `{hostname}` get's replaced with the servers hostname.

This is helpful if you have multiple backend servers behind a load balancer and you want to know on which backend server you are.

Maybe other placeholders make also sense.

**Comment by mholt** [comment]
Placeholders are supported in the `proxy` directive (proxy_header) but not the `header` directive, I guess this is the first request we've had for it.

**Comment by Xe** [comment]
> This is helpful if you have multiple backend servers behind a load balancer and you want to know on which backend server you are.

```go
hostname, _ := os.Hostname()
log.Printf("I'm connected to %s", hostname)
```

**Comment by Burmudar** [comment]
Hi. I'd like to work on this issue as an intro to contributing towards Caddy. Any objections ?

**Comment by abiosoft** [comment]
@Burmudar no objections. No one has claimed this yet, you can go ahead. 👍

**Comment by mholt** [comment]
Closed in #760.
